javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -85,9 +81,80 @@
  * @author Daniel Dietrich
  * @since 1.1.0
  */
-public interface List<T> extends Kind1<List<?>, T>, LinearSeq<T>, Stack<T> {
+public final class List<T> implements Stack<T>, LinearSeq<T>, Kind1<List<?>, T>, Serializable {
```

Comment:
Never mind, I was a little bit shocked because it was the first collection I implemented in Javaslang 1.0. There was already a discussion why it is an interface and not a class but it is intentionally designed that way. It should be `sealed` in order to prevent others to implement it but in Java it is not possible to express it.